pkg/cli/run.go

```diff
@@ -239,6 +240,11 @@ func (s *Run) Run(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
+	// Ensure opts.AutoUpgrade is set if is implied by opts.AutoUpgradeInterval or imageSource.Image's pattern.
+	if opts.AutoUpgrade == nil || !*opts.AutoUpgrade {
+		opts.AutoUpgrade = &[]bool{autoupgrade.ImpliedAutoUpgrade(opts.AutoUpgradeInterval, imageSource.Image)}[0]
+	}
```

Comment:
Quote from @cjellick:
> I think the behavior I want is that the autoUpgrade field must be set, but the CLI helps - if the tag is detected to be a pattern, set the autoUpgrade field for the user. This should be done client side, not server side.

This looks like the "CLI helps" portion, but there is nothing here to require that this field is set.

Reply:
I interpreted the statement that "the autoUpgrade field must be set" to mean that we should require the `autoUpgrade` field be set if the user is trying to use an autoupgrade pattern. Additionally, "but the CLI helps" to mean that if the user hasn't set this field, then the CLI will help them out. I understand that you have done the second part here.

Do we want to require that the `autoUpgrade` field is set whenever an autoupgrade tag is used? For example, as I understand it, any app created outside of the CLI may not satisfy this requirement.

Reply:
> Do we want to require that the autoUpgrade field is set whenever an autoupgrade tag is used?
Yes! That is definitely something we want. I'll code some validation for this into the API Server that will enforce Apps to have `AutoUpgrade` set when its use is assumed.

Reply:
@cjellick Just for context on this discussion.